arch/arc/core/fault.c

```diff
@@ -384,9 +386,9 @@ void _Fault(z_arch_esf_t *esf, uint32_t old_sp)
 		return;
 	}
 
-	LOG_ERR("***** Exception vector: 0x%x, cause code: 0x%x, parameter 0x%x",
+	ARC_EXCEPTION_DUMP("***** Exception vector: 0x%x, cause code: 0x%x, parameter 0x%x",
```

Comment:
All other calls are guarded by CONFIG_EXCEPTION_DEBUG, this one isn't.
That means it's not possible to fully disable the logging/kprintf dependency.
I'd suggest to add some way to disable the printf in case a user doesn't want any logging in a production system or so.
Add the CONFIG_EXCEPTION_DEBUG ifdef here as well?